pkg/kv/kvserver/raftlog/entry.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package raftlog
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/errors"
+	"go.etcd.io/etcd/raft/v3/raftpb"
+)
+
+// Entry contains data related to a raft log entry. This is the raftpb.Entry
+// itself but also all encapsulated data relevant for command application.
+type Entry struct {
+	Meta enginepb.MVCCMetadata // only set after LoadFromRawValue
+	Ent  raftpb.Entry          // must be pre-populated when calling Load
+	// These fields will all be populated by Load.
+	ID  kvserverbase.CmdIDKey         // always set
+	CC1 *raftpb.ConfChange            // only set for config change
+	CC2 *raftpb.ConfChangeV2          // only set for config change
+	CCC *kvserverpb.ConfChangeContext // only set for config change
+	Cmd kvserverpb.RaftCommand        // always set
+}
+
+// LoadFromRawValue populates the Entry from a raw value, i.e. from bytes as
+// stored on a storage.Engine.
+func (e *Entry) LoadFromRawValue(b []byte) error {
+	if err := protoutil.Unmarshal(b, &e.Meta); err != nil {
+		return errors.Wrap(err, "decoding raft log MVCCMetadata")
+	}
+
+	if err := storage.MakeValue(e.Meta).GetProto(&e.Ent); err != nil {
+		return errors.Wrap(err, "unmarshalling raft Entry")
+	}
+
+	return e.load()
+}
+
+// Load populates the Entry from the Ent field. Will reset the Meta field.
+func (e *Entry) Load() error {
```

Comment:
This should take a `raftpb.Entry` and place it in `Ent` instead of callers having to populate it themselves.